### ISSUE TITLE
Property Pane Tabs text colour not following LnF

### ISSUE THREAD

**Issue by damiensellier** [damiensellier/CtrlrX#50]
I tried everything possible to have the text colour of the propery Pane follow the general LnF without success.

setColour() works : 
```
tabbedComponent->getTabbedButtonBar().setColour(TabbedButtonBar::tabTextColourId, Colours::grey);
tabbedComponent->getTabbedButtonBar().setColour(TabbedButtonBar::frontTextColourId, Colours::grey);
```
BUT findColour() won't work : 

```
tabbedComponent->getTabbedButtonBar().setColour(TabbedButtonBar::tabTextColourId, currentLnF.findColour(TabbedButtonBar::tabTextColourId)); 
```

EVEN with getting the default LnF : 
```
auto &currentLnF = owner.getDefaultLookAndFeel(); // Added v5.6.31
tabbedComponent->getTabbedButtonBar().setColour(TabbedButtonBar::tabTextColourId, currentLnF.findColour(TabbedButtonBar::tabTextColourId));
```

This is really strange and probably a JUCE issue.

Although, getting the colour of the tab itself with findColour is working...

`getLookAndFeel().findColour(TabbedComponent::backgroundColourId), `

In context :

```

	CtrlrPanelComponentProperties *props = new CtrlrPanelComponentProperties (owner);  
        
	tabbedComponent->addTab ("General",
                             getLookAndFeel().findColour(TabbedComponent::backgroundColourId), // BG change works but not the tab text.
                             props,
                             true
                             );
    
    tabbedComponent->addTab ("Resources",
                             getLookAndFeel().findColour(TabbedComponent::backgroundColourId), // BG change works but not the tab text.
                             new CtrlrPanelResourceEditor(owner),
                             true
                             );
    
	tabbedComponent->addTab ("Utility",
                             getLookAndFeel().findColour(TabbedComponent::backgroundColourId), // BG change works but not the tab text.
                             new CtrlrPanelUtilities(owner),
                             true
                             );
```

Having the text set to grey and switching tab colour is not ideal since grey on grey or black on black or white on white won't be legible.

For the moment it's kept with fixed colours black text on grey background, but would require further investigation:

![Capture d’écran, le 2024-12-26 à 18 08 57](https://github.com/user-attachments/assets/dc4262c4-7dd9-4932-90fc-56466defcc06)

**Comment by damiensellier** [comment]
CtrlrPanelProperties.h

Requires adding in public:

`void lookAndFeelChanged(); // Added v5.6.31`


As following: 

```
#pragma once

#include "CtrlrPanelFindProperty.h"

class CtrlrPanelEditor;
class CtrlrLookAndFeel;

class CtrlrPanelProperties  : public Component,
                              public ChangeListener,
                              public ChangeBroadcaster
{
public:
    CtrlrPanelProperties (CtrlrPanelEditor &_owner);
    ~CtrlrPanelProperties();
	void changeListenerCallback (ChangeBroadcaster* source);
	void refreshAll();
	void layoutChanged();
    void paint (Graphics& g);
    void resized();
    void lookAndFeelChanged(); // Added v5.6.31
    JUCE_LEAK_DETECTOR(CtrlrPanelProperties)

private:
	CtrlrPanelEditor &owner;
    TabbedComponent* tabbedComponent;
    CtrlrPanelProperties (const CtrlrPanelProperties&);
	std::unique_ptr<CtrlrPanelFindProperty> ctrlrPanelFindProperty;
	const CtrlrPanelProperties& operator= (const CtrlrPanelProperties&);
};

```

CtrlrPanelProperties.cpp

Requires adding `void CtrlrPanelProperties::lookAndFeelChanged()` to force the colour change.

```
void CtrlrPanelProperties::lookAndFeelChanged()
{
    tabbedComponent->getTabbedButtonBar().setColour(TabbedButtonBar::tabTextColourId, findColour(TextEditor::highlightedTextColourId));
    tabbedComponent->getTabbedButtonBar().setColour(TabbedButtonBar::frontTextColourId, findColour(TextEditor::textColourId));
    repaint();
}
```

**Comment by damiensellier** [comment]
FIXED on 26.12.2024 for v5.6.31

CtrlrPanelProperties.cpp & CtrlrPanelProperties.h

![Capture d’écran, le 2024-12-26 à 23 49 27](https://github.com/user-attachments/assets/b81a0725-f562-44aa-9887-ae14bf8042e7)
![Capture d’écran, le 2024-12-26 à 23 49 15](https://github.com/user-attachments/assets/a3768746-f518-44c7-bb23-4698bc4507b1)
![Capture d’écran, le 2024-12-26 à 23 49 06](https://github.com/user-attachments/assets/ddd12a26-fa1f-4391-93ee-e2d60621730e)

**Comment by damiensellier** [comment]
JUCE FORUM DISCUSSION : 
https://forum.juce.com/t/tabbedcomponent-and-tabtextcolourid-not-following-component-lookandfee/59743/2
